packages/perseus/src/widgets/radio/multiple-choice-widget.new.tsx

```diff
@@ -71,278 +71,258 @@ type Props = WidgetProps<
+
+                // Get the reviewChoice from the rubric, if it exists.
+                const reviewChoice = reviewModeRubric?.choices[i];
+
+                return {
+                    id: choice.id,
+                    content: renderContent(content),
+                    checked: selected,
+                    correct:
+                        choice.correct === undefined
+                            ? !!reviewChoice && !!reviewChoice.correct
+                            : choice.correct,
+                    disabled: readOnly,
+                    hasRationale: !!choice.rationale,
+                    rationale: renderContent(choice.rationale),
+                    showRationale: rationaleShown,
+                    showCorrectness: correctnessShown,
+                    isNoneOfTheAbove: !!choice.isNoneOfTheAbove,
+                    revealNoneOfTheAbove: !!(questionCompleted && selected),
+                    previouslyAnswered,
+                };
+            });
+        };
 
-        return choicesProps;
-    };
+        /**
+         * Prepares the choice props for rendering in the MultipleChoiceComponent.
+         *
+         * This function:
+         * 1. Uses getChoiceStates() to determine the appropriate state for each choice
+         *    based on widget mode (review/static) and user selections
+         * 2. Transforms these states into component props via buildChoiceProps(),
+         *    including rendered content, correctness indicators, and rationales
+         *
+         * @returns An array of choice props ready for the component
+         */
+        const prepareChoicesProps = () => {
+            // Get the updated choice states based on the current props
+            const processedChoiceStates = getChoiceStates({
+                choices,
+                isStatic,
+                showSolutions,
+                choiceStates,
+                values,
+            });
+
+            // Build the choice props from the updated choice states
+            return buildChoiceProps(processedChoiceStates);
+        };
 
-    /**
-     * Renders the MultipleChoiceComponent with the current props.
-     *
-     * @returns The rendered React component
-     */
-    render(): React.ReactNode {
-        const choicesProps = this.prepareChoicesProps();
+        const choicesProps = prepareChoicesProps();
+        const numCorrect = props.numCorrect;
 
-        // Extract props that need to be passed to the component
-        const {apiOptions, multipleSelect, countChoices, numCorrect} =
-            this.props;
-        const reviewMode =
-            this.props.reviewMode || this.props.showSolutions === "all";
+        // This is strange, but currently we're showing the same view for both reviewMode and
+        // showSolutions === "all". We may wish to differentiate between the two in the future,
+        // depending on the Perseus GUTC project.
+        const isReviewMode = reviewMode || showSolutions === "all";
```

Comment:
I wanted to call out the strange way we're handling this now, in case we find ourselves confused again in the future. That being said, neither of these variables are particularly clear about how they _should_ be treated differently. I opted not to dig into this too much now, as we'll probably have larger discussions after some progress on GUTC.